internal/storage/vector_chunk_manager.go

```diff
@@ -140,65 +114,30 @@ func (vcm *VectorChunkManager) Exist(filePath string) bool {
 	return vcm.vectorStorage.Exist(filePath)
 }
 
-func (vcm *VectorChunkManager) readWithCache(filePath string) ([]byte, error) {
-	contents, err := vcm.vectorStorage.Read(filePath)
-	if err != nil {
-		return nil, err
-	}
-	results, err := vcm.deserializeVectorFile(filePath, contents)
-	if err != nil {
-		return nil, err
-	}
-	err = vcm.cacheStorage.Write(filePath, results)
-	if err != nil {
-		return nil, err
-	}
-	r, err := vcm.cacheStorage.Mmap(filePath)
-	if err != nil {
-		return nil, err
-	}
-	size, err := vcm.cacheStorage.Size(filePath)
-	if err != nil {
-		return nil, err
-	}
-	vcm.cacheSizeMutex.Lock()
-	vcm.cacheSize += size
-	vcm.cacheSizeMutex.Unlock()
-	if !vcm.fixSize {
-		if vcm.cacheSize < vcm.cacheLimit {
-			if vcm.cache.Len() == vcm.cache.Capacity() {
-				newSize := float32(vcm.cache.Capacity()) * 1.25
-				vcm.cache.Resize(int(newSize))
-			}
-		} else {
-			// +1 is for add current value
-			vcm.cache.Resize(vcm.cache.Len() + 1)
-			vcm.fixSize = true
-		}
-	}
-	vcm.cache.Add(filePath, r)
-	return results, nil
-}
-
 // Read reads the pure vector data. If cached, it reads from local.
 func (vcm *VectorChunkManager) Read(filePath string) ([]byte, error) {
-	if vcm.cacheEnable {
-		if r, ok := vcm.cache.Get(filePath); ok {
-			at := r.(*mmap.ReaderAt)
+	if v, ok := vcm.cache.Get(filePath); ok {
+		if vcm.cache.localCacheEnable {
```

Comment:
why do we need localCacheEnable here?
That makes everything complicated

Reply:
If localFileCache is enabled, we can mmap file into memory. We cache *mmap.ReaderAt in memory.
If localFileCache is not enabled, we cache byte array into memory.
The value v may be []byte or *mmap.ReadAt, so the complexity is unavoidable.